### PR TITLE
More depth/stencil renderbuffer initialization tests

### PR DESCRIPTION
**Pull request by kainino0x**
Both multisampled for WebGL 2.0 and non-multisampled for WebGL 1.0.

### CODE REVIEW

**Comment by kainino0x** [comment]
I've been meaning to write these tests ever since #2418. Just finally got around to it because I was looking for a small task to get done today. @zhenyao PTAL

File: sdk/tests/conformance/renderbuffers/depth-renderbuffer-initialization.html [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR,
+            "should be no error after renderbufferStorage.");
+    }
+
+    shouldBe("gl.checkFramebufferStatus(gl.FRAMEBUFFER)",
+             "gl.FRAMEBUFFER_COMPLETE");
+    wtu.glErrorShouldBe(gl, gl.NO_ERROR, 'should be no errors');
+
+    // Draw a blue quad (at z = 0.0, so depth = 0.5) (should pass the depth test)
+    gl.depthFunc(gl.LESS);
```
Comment (zhenyao) [inline sdk/tests/conformance/renderbuffers/depth-renderbuffer-initialization.html]: The current depth should be initialized to 0? so depth == 0.5 with gl.LESS should not pass the depth test actually. Or am I missing something here?

File: sdk/tests/conformance/renderbuffers/stencil-renderbuffer-initialization.html [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    }
+    gl.bindRenderbuffer(gl.RENDERBUFFER, buffer);
+    gl.framebufferRenderbuffer(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.RENDERBUFFER, buffer);
+    gl.bindRenderbuffer(gl.RENDERBUFFER, stencilBuffer);
+    gl.framebufferRenderbuffer(gl.FRAMEBUFFER, gl.STENCIL_ATTACHMENT, gl.RENDERBUFFER, stencilBuffer);
+    if (params.alloc2) {
+        if (params.alloc1) {
+            // Clear the FBO in order to make sure framebufferRenderbuffer is
+            // committed.
+            gl.clear(gl.COLOR_BUFFER_BIT | gl.STENCIL_BUFFER_BIT);
```
Comment (zhenyao) [inline sdk/tests/conformance/renderbuffers/stencil-renderbuffer-initialization.html]: You should check for framebuffer completeness before clear.

File: sdk/tests/conformance/renderbuffers/stencil-renderbuffer-initialization.html [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    }
+
+    function allocStorage(format, width, height) {
+        gl.renderbufferStorage(gl.RENDERBUFFER, format, width, height);
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR,
+            "should be no error after renderbufferStorage.");
+    }
+
+    shouldBe("gl.checkFramebufferStatus(gl.FRAMEBUFFER)",
+             "gl.FRAMEBUFFER_COMPLETE");
```
Comment (zhenyao) [inline sdk/tests/conformance/renderbuffers/stencil-renderbuffer-initialization.html]: I am not sure if this is required. You should just return early without failing the test if fbo is incomplete here.

File: sdk/tests/conformance2/renderbuffers/multisampled-stencil-renderbuffer-initialization.html [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    gl.bindFramebuffer(gl.FRAMEBUFFER, fbo);
+    var buffer = gl.createRenderbuffer();
+    gl.bindRenderbuffer(gl.RENDERBUFFER, buffer);
+    gl.renderbufferStorage(gl.RENDERBUFFER, gl.RGBA8, resolve.w, resolve.h);
+    gl.framebufferRenderbuffer(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.RENDERBUFFER, buffer);
+    var stencilBuffer = gl.createRenderbuffer();
+    gl.bindRenderbuffer(gl.RENDERBUFFER, stencilBuffer);
+    gl.renderbufferStorage(gl.RENDERBUFFER, gl.STENCIL_INDEX8, resolve.w, resolve.h);
+    gl.framebufferRenderbuffer(gl.FRAMEBUFFER, gl.STENCIL_ATTACHMENT, gl.RENDERBUFFER, stencilBuffer);
+    shouldBe("gl.checkFramebufferStatus(gl.FRAMEBUFFER)",
```
Comment (zhenyao) [inline sdk/tests/conformance2/renderbuffers/multisampled-stencil-renderbuffer-initialization.html]: Same here, return early if incomplete instead

File: sdk/tests/conformance2/renderbuffers/multisampled-stencil-renderbuffer-initialization.html [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+
+    function allocStorage(width, height, samples) {
+        gl.renderbufferStorageMultisample(
+            gl.RENDERBUFFER, samples, gl.STENCIL_INDEX8, width, height);
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR,
+            "should be no error after renderbufferStorageMultisample(STENCIL_INDEX8).");
+    }
+
+    if (gl.checkFramebufferStatus(gl.FRAMEBUFFER) != gl.FRAMEBUFFER_COMPLETE) {
+        debug("Skip: Stencil-only framebuffer is allowed to be incomplete.");
```
Comment (zhenyao) [inline sdk/tests/conformance2/renderbuffers/multisampled-stencil-renderbuffer-initialization.html]: note here you did the right thing, so you should do the same everywhere else.

File: sdk/tests/conformance2/renderbuffers/multisampled-stencil-renderbuffer-initialization.html [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+    // Blit from multisampled buffer to non-multisampled buffer.
+    gl.bindFramebuffer(gl.READ_FRAMEBUFFER, fbo_m);
+    gl.bindFramebuffer(gl.DRAW_FRAMEBUFFER, fbo);
+    gl.clearStencil(2);
+    gl.clear(gl.STENCIL_BUFFER_BIT);
+    // Blit stencil from fbo_m (should be 0) to fbo (should be 2).
+    gl.blitFramebuffer(0, 0, resolve.w, resolve.h,
+                       0, 0, resolve.w, resolve.h,
+                       gl.STENCIL_BUFFER_BIT, gl.NEAREST);
+    // fbo stencil should now be 1.0.
```
Comment (zhenyao) [inline sdk/tests/conformance2/renderbuffers/multisampled-stencil-renderbuffer-initialization.html]: should now be 0.0?

File: sdk/tests/conformance/renderbuffers/depth-renderbuffer-initialization.html (hunk 1 above)
Comment (jdashg) [reply sdk/tests/conformance/renderbuffers/depth-renderbuffer-initialization.html]: Please use 'depth' to refer to the [0-1] depth range, and 'clip z' to refer to [-1,1] clip space coordinates.

**Comment by kainino0x** [comment]
I forgot this was still open. Looking into revisions now.

File: sdk/tests/conformance/renderbuffers/depth-renderbuffer-initialization.html (hunk 1 above)
Comment (kainino0x) [reply sdk/tests/conformance/renderbuffers/depth-renderbuffer-initialization.html]: When depth is uninitialized, it defaults to 1.0. So the depth test should pass.

**Comment by kainino0x** [comment]
Done.

**Comment by zhenyao** [comment]
LGTM
Thanks

**Pull request merged**